### PR TITLE
TASK-LINKS-3, added display of chosen cathegory, smaller UI changes

### PR DESCRIPTION
**Pull request by FilipKalicanin**

### CODE REVIEW

File: src/displayUI.js [hunk 1]
```diff
@@ -18,15 +19,16 @@ export function displayOneCategory(categoryElement) {
...
-  categoryItem.addEventListener("click", (e) => {
+  category.addEventListener("click", (e) => {
     e.preventDefault();
     document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
     let firstWord = categoryElement.name;
     let parameter = firstWord.split(" ")[0];
-    instanceOfMainClass.displayLinks(parameter);
+    instanceOfMainClass.collectAllLinks(parameter);
     instanceOfMainClass.setTextFilter(parameter);
+    selectedCategoryDisplay(firstWord);
```
Comment (markoprogrammer) [inline src/displayUI.js:30]: What is purpose of selectedCategoryDisplay?
When something selected is enough style/class change.

Comment (FilipKalicanin) [reply src/displayUI.js:30]: I tought you said that beside it is highlited, is should be visible under all categories as well, just like on that car site that we saw.

File: src/mainClass.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 
   constructor(searchCategory, searchLinks, displayLinksForChosenCategory) {
+    // categories, search functionality
     this.categories = [];
     this.onReceivedCategories = searchCategory;
-    this.displayChosenLinks = displayLinksForChosenCategory;
+    // links, search functionality and initial display when app starts
     this.links = [];
     this.onReceivedLinks = searchLinks;
+    this.displayChosenLinks = displayLinksForChosenCategory;
```
Comment (markoprogrammer) [inline src/mainClass.js]: displayChosenLinks, we don't need since we already have onReceived callback, and getAllLinks

File: src/mainClass.js [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+    // localStorage
     this.storageLinks = JSON.parse(localStorage.getItem('links'));
     this.storageCategories = JSON.parse(localStorage.getItem('categories'));
+    // used for search
```
Comment (markoprogrammer) [inline src/mainClass.js:15]: Make sure, that you accomplished workflow, that we defined with receive data method, receive callback, getAll and filter text. For both categories and links. So to we have simplified handlings.
So UI could  initialise load of data (for links with/or without category, for categories always same without param).
When user search data text is set as property, and called getAll to receive filtered array.
When clicked on category is called method with param to receive links again (same method as initial load), and when callback called (data received) is called getAll which will return links for rendering. 
Same when filter text changed, we set text prop, and call getAll again to receive property.
So basically we use method to load data, callback to notify ui that data is received and ui could call getAll and render. When we filter by text we set prop text and call getAll to receive data.

File: src/displayUI.js (hunk 1 above)
Comment (markoprogrammer) [reply src/displayUI.js:30]: Ok but every UI state, should be covered in model, so we need to have information what is selected category so to we know our state, and we could rerender ui.
